replica/table.cc

```diff
@@ -550,17 +550,28 @@ void table::enable_off_strategy_trigger() {
 
 storage_group_manager::~storage_group_manager() = default;
 
-future<> storage_group_manager::parallel_foreach_storage_group(std::function<future<>(size_t, storage_group&)> f) {
-    rwlock::holder shared_lock = co_await get_rwlock().hold_read_lock();
-    co_await coroutine::parallel_for_each(_storage_groups, [&] (std::pair<size_t, const std::unique_ptr<storage_group>&> p) -> future<> {
-        co_await f(p.first, *p.second.get());
+// exception-less attempt to hold gate.
+static std::optional<gate::holder> try_hold_gate(gate& g) noexcept {
```

Comment:
Can be added to seastar as `gate::try_hold` as a follow up.

Reply:
https://github.com/scylladb/seastar/pull/2315

Reply:
thanks.